modbusrtumaster.cpp

```diff
@@ -0,0 +1,274 @@
+                break;
+            case QModbusDataUnit::RegisterType::InputRegisters:
+                emit receivedInputRegister(reply->serverAddress(), modbusAddress, unit.value(i));
+                break;
+            case QModbusDataUnit::RegisterType::HoldingRegisters:
+                emit receivedHoldingRegister(reply->serverAddress(), modbusAddress, unit.value(i));
+                break;
+            case QModbusDataUnit::RegisterType::Invalid:
+                break;
+            }
+        }
+
+    } else if (reply->error() == QModbusDevice::ProtocolError) {
+        qCWarning(dcModbusCommander()) << "Read response error:" << reply->errorString() << reply->rawResult().exceptionCode();
+    } else {
+        qCWarning(dcModbusCommander()) << "Read response error:" << reply->error();
+    }
+    reply->deleteLater();
+}
+
+void ModbusRTUMaster::onReplyErrorOccured(QModbusDevice::Error error)
+{
+    qCWarning(dcModbusCommander()) << "Modbus replay error:" << error;
+    QModbusReply *reply = qobject_cast<QModbusReply *>(sender());
+    if (!reply)
+        return;
+    reply->finished(); //to make sure it will be deleted
+}
+
+void ModbusRTUMaster::onReconnectTimer()
+{
+    if(!m_modbusRtuSerialMaster->connectDevice()) {
+        m_reconnectTimer->start(10000);
+    }
+}
+
+bool ModbusRTUMaster::readCoil(int slaveAddress, int registerAddress)
+{
+    if (!m_modbusRtuSerialMaster)
+        return false;
+
+    QModbusDataUnit request = QModbusDataUnit(QModbusDataUnit::RegisterType::Coils, registerAddress, 1);
+
+    if (QModbusReply *reply = m_modbusRtuSerialMaster->sendReadRequest(request, slaveAddress)) {
+        if (!reply->isFinished()) {
+            connect(reply, &QModbusReply::finished, this, &ModbusRTUMaster::onReplyFinished);
+            connect(reply, &QModbusReply::errorOccurred, this, &ModbusRTUMaster::onReplyErrorOccured);
+            QTimer::singleShot(200, reply, SLOT(deleteLater()));
+        } else {
+            delete reply; // broadcast replies return immediately
+        }
+    } else {
+        qCWarning(dcModbusCommander()) << "Read error: " << m_modbusRtuSerialMaster->errorString();
+    }
+    return true;
+}
+
+bool ModbusRTUMaster::writeCoil(int slaveAddress, int registerAddress, bool value)
+{
+    if (!m_modbusRtuSerialMaster)
+        return false;
+
+    QModbusDataUnit request = QModbusDataUnit(QModbusDataUnit::RegisterType::Coils, registerAddress, 1);
+    request.setValue(0, static_cast<uint16_t>(value));
+
+    if (QModbusReply *reply = m_modbusRtuSerialMaster->sendWriteRequest(request, slaveAddress)) {
+        if (!reply->isFinished()) {
+            connect(reply, &QModbusReply::finished, this, &ModbusRTUMaster::onReplyFinished);
+            connect(reply, &QModbusReply::errorOccurred, this, &ModbusRTUMaster::onReplyErrorOccured);
+            QTimer::singleShot(200, reply, SLOT(deleteLater()));
```

Comment:
you can use also here the new notiation: `QTimer::singleShot(200, reply, &QModbusReply::deleteLater);`

Applies also for the other single shots...